### DISCUSSION TITLE
Suggestion of java (slf4j) check

### DISCUSSION THREAD

**Discussion by selliera** [pmd/pmd#5063] [Ideas]
Hello,

In SLF4J 2, which is a very common and widespread java logging facade api, there are functions that are meant to fluently create new logging statements.
https://www.slf4j.org/api/org/slf4j/Logger.html see atDebug, atError, atWarning, atInfo, atTrace, atLevel

Those are very convenient, and allow to use closures to compute what to display only if the message has to be logged.
Common usage is like:

```java
    logger.atError().setMessage("some message with two arguments {} ms: {}")
          .addArgument("some string argument")
          .addArgument(() -> System.currentTimeMillis() - startDate)
          .log();
```

A very common error is to forget the ending `log()` call.

With code like:
```java
    logger.atError().setMessage("some message with two arguments {} ms: {}")
          .addArgument("some string argument")
          .addArgument(() -> System.currentTimeMillis() - startDate);
```
you can spend an enormous amount of time wondering why your expected logging statement is not in the output.

If PMD could detect that the `org.slf4j.spi.LoggingEventBuilder` returned by the last `addArgument()` is neither used by `log()` nor stored in a variable, we could catch this kind of errors before runtime.

**Comment by UncleOwen** [comment]
PMD 7.27.0 will include a new rule [UnusedReturnValue](https://docs.pmd-code.org/snapshot/pmd_rules_java_errorprone.html#unusedreturnvalue) which will do this. Thanks for the suggestion!
